src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -936,4 +937,47 @@ public String getMappedColumnName() {
         }
     }
 
+    /**
+     * This method is used to validate the dates of an exercise. A date is valid if there is no dueDateError or assessmentDueDateError
+     * @throws BadRequestException if the dates are not valid
+     */
+    public void validateDates() throws BadRequestException {
```

Comment:
```suggestion
    public void validateDates() throws BadRequestAlertException {
```

Reply:
or you might be able to leave it out, but BadRequestException is definitely wrong here